src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseChecklistService.cs

```diff
@@ -101,9 +104,50 @@ public async Task<List<ReleaseChecklistIssue>> GetErrors(Release release)
                 errors.Add(new ReleaseChecklistIssue(ValidationErrorMessages.ReleaseNoteRequired));
             }
 
+            if (await ReleaseHasEmptyGenericContentSection(release.Id))
+            {
+                errors.Add(new ReleaseChecklistIssue(ValidationErrorMessages.EmptyContentSectionExists));
+            }
+
+            if (ReleaseGenericContentSectionsContainEmptyContentBlock(release.Id))
+            {
+                errors.Add(new ReleaseChecklistIssue(ValidationErrorMessages.GenericSectionsContainEmptyHtmlBlock));
+            }
+
             return errors;
         }
 
+        private async Task<bool> ReleaseHasEmptyGenericContentSection(Guid releaseId)
+        {
+            return await _contentDbContext.ReleaseContentSections
+                .Include(rcs => rcs.ContentSection)
+                .ThenInclude(cs => cs.Content)
+                .Where(rcs =>
+                    rcs.ReleaseId == releaseId
+                    && rcs.ContentSection.Type == ContentSectionType.Generic)
+                .AnyAsync(rcs => rcs.ContentSection.Content.Count == 0);
+        }
+
+        private bool ReleaseGenericContentSectionsContainEmptyContentBlock(Guid releaseId)
```

Comment:
This method returns `bool` whilst the other returns `Task<bool>`. Would just neaten these up by making them have consistent return types either way.